src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,24 +1,92 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 /**
  * <p>Реалізувати свою HashMap, а саме методи `put(K key, V value)`, `getValue()` та `getSize()`.
  * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 1 << 4;
+    private static final double FILLFACTOR = 0.75;
+
+    private Entry<K, V>[] buckets;
+    private int size;
+    private double trashHold;
+
+    public MyHashMap() {
+        buckets = new Entry[INITIAL_CAPACITY];
+        trashHold = INITIAL_CAPACITY * FILLFACTOR;
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size == trashHold) {
+            resize();
+        }
+        int index = getIndex(key, buckets.length);
+        Entry<K, V> checkEntry = getEntry(key, index);
 
+        if (checkEntry != null) {
+            checkEntry.value = value;
+        } else {
+            Entry<K, V> node = new Entry<>(key, value, buckets[index]);
+            buckets[index] = node;
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Entry<K, V> entry = getEntry(key, getIndex(key, buckets.length));
+        return entry != null ? entry.value : null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key, int capacity) {
```

Comment:
```suggestion
    private int getIndex(K key) {
```
this method always takes `buckets.length` so I think, there is no need to make it as a parameter

Reply:
done